### ISSUE TITLE
bug: enforced timeout should return 504 gateway timeout

### ISSUE THREAD

**Issue by asoorm** [TykTechnologies/tyk#1975]
**Do you want to request a *feature* or report a *bug*?**

bug

**What is the current behavior?**

When `Enforced Timeout` plugin is enabled, the gateway response status code indicates a client error (408 Request Timeout).

```
curl http://gateway/httpbin/delay/3 -v
> GET /httpbin/delay/3 HTTP/1.1
> Host: mckinsey-1
> User-Agent: curl/7.61.0
> Accept: */*
>
< HTTP/1.1 408 Request Timeout
< Content-Type: application/json
< Date: Sat, 10 Nov 2018 01:26:54 GMT
< Content-Length: 57
<
{
    "error": "Upstream service reached hard timeout."
}
```

**What is the expected behavior?**

The gateway should return the correct status code. `504 - Gateway Timeout`.

> The server was acting as a gateway or proxy and did not receive a timely response from the upstream server.

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem**

Reverse proxy to httpbin.org
Configure an enforced timeout plugin listening on path `delay/{seconds}`, and apply a timeout of 3s
Send a call to say `delay/4` ensuring the upstream service will respond after 4s.

**Which versions of Tyk affected by this issue? Did this work in previous versions of Tyk?**

All?

**Comment by ilijabojanovic** [comment]
Verified

**Comment by buger** [comment]
https://github.com/TykTechnologies/tyk/pull/1977 was merged to `release-2.7` branch
<details><summary></summary>Created via API</details>

**Comment by buger** [comment]
https://github.com/TykTechnologies/tyk/pull/1977 was merged to `release-2.8` branch
<details><summary></summary>Created via API</details>
